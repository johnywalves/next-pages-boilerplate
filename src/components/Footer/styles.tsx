import styled from 'styled-components'

export const WrapperFooter = styled.div`
  padding: 2rem;

  & span {
    color: red;
  }

  & a {
    color: var(--color-primary);

    &:hover {
      text-decoration: underline;
    }
  }
`
