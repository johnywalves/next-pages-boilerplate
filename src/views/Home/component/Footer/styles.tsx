import styled from 'styled-components'

export const WrapperFooter = styled.div`
  padding: 2rem;
  color: var(--color-neutral-100);

  & span {
    color: var(--color-danger);
  }

  & a {
    color: var(--color-primary-pure);

    &:hover {
      text-decoration: underline;
    }
  }
`
