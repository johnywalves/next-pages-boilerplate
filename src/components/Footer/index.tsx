import Link from 'next/link'

import { WrapperFooter } from './styles'

const Footer = () => (
  <WrapperFooter>
    <p>
      Feito com <span>â™¥</span> por{' '}
      <Link href="https://johnywalves.com.br/">Johny W. Alves</Link>
    </p>
  </WrapperFooter>
)

export default Footer
