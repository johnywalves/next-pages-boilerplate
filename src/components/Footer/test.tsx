import { render } from '@testing-library/react'

import Footer from '.'

describe('<Footer />', () => {
  it('should render', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
