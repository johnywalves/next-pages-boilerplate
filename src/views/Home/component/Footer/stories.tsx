import { Meta, StoryObj } from '@storybook/react'

import Footer from '.'

const meta: Meta<typeof Footer> = {
  title: 'Footer',
  component: Footer,
  tags: ['autodocs'],
  args: {}
}

type Story = StoryObj<typeof Footer>

export const Primary: Story = {
  args: {}
}

export default meta
