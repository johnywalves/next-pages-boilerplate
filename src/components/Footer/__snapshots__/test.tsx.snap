// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c0 {
  padding: 2rem;
}

.c0 span {
  color: red;
}

.c0 a {
  color: var(--color-primary);
}

.c0 a:hover {
  text-decoration: underline;
}

<div
  class="c0"
>
  <p>
    Feito com 
    <span>
      â™¥
    </span>
     por
     
    <a
      href="https://johnywalves.com.br/"
    >
      Johny W. Alves
    </a>
  </p>
</div>
`;
